src/pgproc.rs

```diff
@@ -0,0 +1,98 @@
+use pgx::{pg_sys, FromDatum, IntoDatum};
+use std::mem::ManuallyDrop;
+use std::ptr::NonNull;
+
+/// Provides a safe wrapper around a Postgres "SysCache" entry from `pg_catalog.pg_proc`.
+pub(crate) struct PgProc {
+    inner: ManuallyDrop<NonNull<pg_sys::HeapTupleData>>,
```

Comment:
This is the only thing I did that I ain't too sure about.  Dunno if ManuallyDrop is required here since below on line 10 Drop is implemented for `PgProc`.

Reply:
Last I checked, Rust doesn't attempt to run Drop on things behind pointers, so ManuallyDrop is indeed redundant here (but if there's a need to carefully handle HeapTupleData's fields you should note that pervasively, obv).

Reply:
Not the fields per se but the pointer does need to be released.  I’ll circle back and clean this up. Don’t think it’s worth holding up a merge. Thanks!